core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -76,6 +76,9 @@ public PartitionData copy() {
   private byte[] keyMetadata = null;
   private Integer sortOrderId;
 
+  private Long dataSequenceNumber = null;
```

Comment:
Is there any particular reason to have these two as a separate block? My initial thought was to put it somewhere next to `partitionSpecId` above.

Reply:
I didn't realise that pattern with the blocks because 'avroSchema' and 'fromProjectionPos/partitionType' also make their own block. Regardless, I merged the new ones with the optional block.